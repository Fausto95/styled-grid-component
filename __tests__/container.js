import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Flex from '../src/flex';

test('it works', () => {
  const tree = renderer.create(<Flex />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('display', 'flex');
});

test('it works - rowReverse', () => {
  const tree = renderer.create(<Flex rowReverse />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('flex-direction', 'row-reverse');
});

test('it works - column', () => {
  const tree = renderer.create(<Flex column />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('flex-direction', 'column');
});

test('it works - columnReverse', () => {
  const tree = renderer.create(<Flex columnReverse />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('flex-direction', 'column-reverse');
});

test('it works - wrapReverse', () => {
  const tree = renderer.create(<Flex wrapReverse />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('flex-wrap', 'wrap-reverse');
});

test('it works - wrap', () => {
  const tree = renderer.create(<Flex wrap />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('flex-wrap', 'wrap');
});

test('it works - alignCenter', () => {
  const tree = renderer.create(<Flex alignCenter />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('align-items', 'center');
});

test('it works - alignStart', () => {
  const tree = renderer.create(<Flex alignStart />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('align-items', 'flex-start');
});

test('it works - alignEnd', () => {
  const tree = renderer.create(<Flex alignEnd />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('align-items', 'flex-end');
});

test('it works - alignBaseline', () => {
  const tree = renderer.create(<Flex alignBaseline />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('align-items', 'baseline');
});

test('it works - alignStretch', () => {
  const tree = renderer.create(<Flex alignStretch />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('align-items', 'stretch');
});

test('it works - inline', () => {
  const tree = renderer.create(<Flex inline />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('display', 'inline-flex');
});

test('it works - contentCenter', () => {
  const tree = renderer.create(<Flex contentCenter />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('align-content', 'center');
});

test('it works - contentStart', () => {
  const tree = renderer.create(<Flex contentStart />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('align-content', 'flex-start');
});

test('it works - contentEnd', () => {
  const tree = renderer.create(<Flex contentEnd />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('align-content', 'flex-end');
});

test('it works - contentBaseline', () => {
  const tree = renderer.create(<Flex contentBaseline />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('align-content', 'baseline');
});

test('it works - alignStart', () => {
  const tree = renderer.create(<Flex alignStart />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('align-items', 'flex-start');
});

test('it works - contentStretch', () => {
  const tree = renderer.create(<Flex contentStretch />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('align-content', 'stretch');
});

test('it works - contentAround', () => {
  const tree = renderer.create(<Flex contentAround />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('align-content', 'around');
});

test('it works - full', () => {
  const tree = renderer.create(<Flex full />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('width', '100%');
  expect(tree).toHaveStyleRule('height', '100%');
  expect(tree).toHaveStyleRule('flex-basis', '100%');
});

test('it works - center', () => {
  const tree = renderer.create(<Flex center />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('align-items', 'center');
  expect(tree).toHaveStyleRule('justify-content', 'center');
});

test('it works - justifyCenter', () => {
  const tree = renderer.create(<Flex justifyCenter />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('justify-content', 'center');
});

test('it works - justifyStart', () => {
  const tree = renderer.create(<Flex justifyStart />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('justify-content', 'flex-start');
});

test('it works - justifyEnd', () => {
  const tree = renderer.create(<Flex justifyEnd />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('justify-content', 'flex-end');
});

test('it works - justifyBetween', () => {
  const tree = renderer.create(<Flex justifyBetween />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('justify-content', 'space-between');
});

test('it works - justifyAround', () => {
  const tree = renderer.create(<Flex justifyAround />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('justify-content', 'space-around');
});

test('it works - justifyEvenly', () => {
  const tree = renderer.create(<Flex justifyEvenly />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('justify-content', 'space-evenly');
});
