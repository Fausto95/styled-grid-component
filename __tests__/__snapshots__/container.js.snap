// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works - alignBaseline 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

<div
  className="c0"
/>
`;

exports[`it works - alignCenter 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
/>
`;

exports[`it works - alignEnd 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

<div
  className="c0"
/>
`;

exports[`it works - alignStart 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

<div
  className="c0"
/>
`;

exports[`it works - alignStart 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

<div
  className="c0"
/>
`;

exports[`it works - alignStretch 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

<div
  className="c0"
/>
`;

exports[`it works - center 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
/>
`;

exports[`it works - column 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
/>
`;

exports[`it works - columnReverse 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
}

<div
  className="c0"
/>
`;

exports[`it works - contentAround 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: around;
  -ms-flex-line-pack: around;
  align-content: around;
}

<div
  className="c0"
/>
`;

exports[`it works - contentBaseline 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: baseline;
  -ms-flex-line-pack: baseline;
  align-content: baseline;
}

<div
  className="c0"
/>
`;

exports[`it works - contentCenter 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
}

<div
  className="c0"
/>
`;

exports[`it works - contentEnd 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: flex-end;
  -ms-flex-line-pack: flex-end;
  align-content: flex-end;
}

<div
  className="c0"
/>
`;

exports[`it works - contentStart 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
}

<div
  className="c0"
/>
`;

exports[`it works - contentStretch 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
}

<div
  className="c0"
/>
`;

exports[`it works - full 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
}

<div
  className="c0"
/>
`;

exports[`it works - inline 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<div
  className="c0"
/>
`;

exports[`it works - justifyAround 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

<div
  className="c0"
/>
`;

exports[`it works - justifyBetween 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div
  className="c0"
/>
`;

exports[`it works - justifyCenter 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
/>
`;

exports[`it works - justifyEnd 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

<div
  className="c0"
/>
`;

exports[`it works - justifyEvenly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

<div
  className="c0"
/>
`;

exports[`it works - justifyStart 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

<div
  className="c0"
/>
`;

exports[`it works - rowReverse 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

<div
  className="c0"
/>
`;

exports[`it works - wrap 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<div
  className="c0"
  wrap={true}
/>
`;

exports[`it works - wrapReverse 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap-reverse;
  -ms-flex-wrap: wrap-reverse;
  flex-wrap: wrap-reverse;
}

<div
  className="c0"
/>
`;

exports[`it works 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="c0"
/>
`;
